### PR TITLE
Add test and improve readme

### PR DESCRIPTION
**Pull request by iislucas**
* Fix `run.*.sh` script references in docs, and clarifying and add a few other links and TODOs to the README.
* Add a simple test for the custom ROC-AUC computation.

### CODE REVIEW

File: experiments/README.md [hunk 1]
```diff
@@ -14,18 +27,55 @@ Install library dependencies (it is optional, but recommended to install these i
...
-To train an existing model,  simply execute the command `./tf_trainer/MODEL_NAME/run.sh` (to run locally) or the command `./tf_trainer/MODEL_NAME/run_ml_engine.sh` to run training on [Google ML Engine](https://cloud.google.com/ml-engine/).
+To train an existing model, execute either command:
+ * `./tf_trainer/MODEL_NAME/run.local.sh` to run training locally, or
+ * `./tf_trainer/MODEL_NAME/run.ml_engine.sh` to run training on [Google ML
+Engine](https://cloud.google.com/ml-engine/).
+
+These scripts assume that you have access to the resources on our cloud
+projects. If you don't, you can still run the models locally, but will have to
+modify the data paths in `run.local.sh`. At the moment, we only support reading
+data in `tf.record` format.
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/README.md]: cool! where is that?

[the remark this answers is not in the document]
Comment (nthain) [reply experiments/README.md]: https://github.com/conversationai/conversationai-models/blob/master/experiments/tf_trainer/common/convert_csv_to_tfrecord.py

File: experiments/tf_trainer/common/base_keras_model_test.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  def test_roc_auc(self):
+    y_true = tf.convert_to_tensor([0.0, 0.7, 0.3, 1.0, 0.2, 1.0, 0.0], dtype=tf.float32)
+    y_pred_05_auc = tf.convert_to_tensor([0.2, 0.8, 0.2, 0.9, 1.0, 0.1, 0.5], dtype=tf.float32)
+    y_pred_08_auc = tf.convert_to_tensor([0.2, 0.8, 0.2, 0.9, 1.0, 1.0, 0.5], dtype=tf.float32)
+
+    auc_05 = base_keras_model.BaseKerasModel.roc_auc(y_true, y_pred_05_auc, threshold=0.5)
+    auc_08 = base_keras_model.BaseKerasModel.roc_auc(y_true, y_pred_08_auc, threshold=0.5)
+
+    init_g = tf.global_variables_initializer()
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/tf_trainer/common/base_keras_model_test.py]: I was wondering about this too... I think I'd like to have a setup for simpler testing where you don't have to think about these kinds of complexities for most tests, so I've moved it the context setup into into a separate utility function which does initialization, so we don't need this in the actual test.

File: experiments/tf_trainer/common/base_keras_model_test.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import numpy as np
+import tensorflow as tf
+
+
+class BaseKerasModelTest(tf.test.TestCase):
+
+  def test_roc_auc(self):
+    y_true = tf.convert_to_tensor([0.0, 0.7, 0.3, 1.0, 0.2, 1.0, 0.0], dtype=tf.float32)
+    y_pred_05_auc = tf.convert_to_tensor([0.2, 0.8, 0.2, 0.9, 1.0, 0.1, 0.5], dtype=tf.float32)
+    y_pred_08_auc = tf.convert_to_tensor([0.2, 0.8, 0.2, 0.9, 1.0, 1.0, 0.5], dtype=tf.float32)
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/tf_trainer/common/base_keras_model_test.py]: good idea. Done.

File: experiments/tf_trainer/common/base_keras_model.py [hunk 4]
```diff
@@ -57,7 +57,8 @@ def new_model_fn(features, labels, mode, params, config):
   @staticmethod
   def roc_auc(y_true: types.Tensor, y_pred: types.Tensor,
               threshold=0.5) -> types.Tensor:
-    """ROC AUC based on TF's metrics package.
+    """ROC AUC based on TF's metrics package. This provide AUC in a Keras
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/tf_trainer/common/base_keras_model.py]: done.

File: experiments/README.md [hunk 5]
```diff
@@ -14,18 +27,55 @@ Install library dependencies (it is optional, but recommended to install these i
...
+
 ## Training an Existing Model
 
-To train an existing model,  simply execute the command `./tf_trainer/MODEL_NAME/run.sh` (to run locally) or the command `./tf_trainer/MODEL_NAME/run_ml_engine.sh` to run training on [Google ML Engine](https://cloud.google.com/ml-engine/).
+To train an existing model, execute either command:
+ * `./tf_trainer/MODEL_NAME/run.local.sh` to run training locally, or
+ * `./tf_trainer/MODEL_NAME/run.ml_engine.sh` to run training on [Google ML
+Engine](https://cloud.google.com/ml-engine/).
+
+These scripts assume that you have access to the resources on our cloud
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/README.md]: These TODOs are above.

File: experiments/README.md [hunk 6]
```diff
@@ -1,10 +1,23 @@
...
+
+This directory contains amn ML framework for text classification. We illustrate
+it with toxic (and other attributes) comment classification.
+
+The framework is structured as a series of common files and templates to quickly
+construct models on top of the [Keras](https://keras.io/) or the [TensorFlow
+Estimator API](https://www.tensorflow.org/programmers_guide/estimators).
+
+The templates also demonstrate how these models can be trained using [Google ML
+Engine](https://cloud.google.com/ml-engine/) and experiments managed with
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/README.md]: tweaked.

File: experiments/README.md [hunk 7]
```diff
@@ -1,10 +1,23 @@
-# Toxic Classification Framework
+# Text Classification Framework
+
+This directory contains amn ML framework for text classification. We illustrate
```
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/README.md]: done.

File: experiments/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (iislucas) [reply experiments/README.md]: Thanks, added comment with a link to it.

**Pull request merged**